Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -22,922 +22,340 @@
 #define FACECONVERTER_H
 
 #include "CarveHeaders.h"
-
 #include <memory>
-
 #include "ConverterBase.h"
-
 #include "CurveConverter.h"
 #include "PlacementConverter.h"
 #include "GeomUtils.h"
 #include "GeometryInputData.h"
 #include "PlacementConverter.h"
-
 #include "BlueFramework/Core/Diagnostics/log.h"
 
-
 namespace OpenInfraPlatform {
 	namespace Core {
 		namespace IfcGeometryConverter {
 			class ItemData;
-
+			
+			/*! \brief Converter functionality for \c IfcSurfaces's subtypes.
+			*
+         	* This class includes converter functions for
+            * \c IfcBoundedSurface and its subtypes,
+            * \c IfcElementarySurface and its subtypes and
+          	* \c IfcSweptSurface.	
+            *
+            * \param IfcEntityTypesT The IFC version templates
+            */
+			
 			template <
 				class IfcEntityTypesT
 			>
-			class FaceConverterT : public ConverterBaseT<IfcEntityTypesT>
+				class FaceConverterT : public ConverterBaseT<IfcEntityTypesT>
 			{
 			public:
-				FaceConverterT(std::shared_ptr<GeometrySettings> geomSettings,
-					std::shared_ptr<UnitConverter<IfcEntityTypesT>> unitConverter,
-					std::shared_ptr<PlacementConverterT<IfcEntityTypesT>> pc,
-					std::shared_ptr<CurveConverterT<IfcEntityTypesT>> cc)
-					: 
+
+				//! Constructor
+				FaceConverterT(EXPRESSReference<GeometrySettings> geomSettings,
+					EXPRESSReference<UnitConverter<IfcEntityTypesT>> unitConverter,
+					EXPRESSReference<PlacementConverterT<IfcEntityTypesT>> pc,
+					EXPRESSReference<CurveConverterT<IfcEntityTypesT>> cc)
+					:
 					ConverterBaseT<IfcEntityTypesT>(geomSettings, unitConverter),
 					placementConverter(pc),
 					curveConverter(cc)
-				{
-				}
+					{
+					}
 
+				//! Virtual destructor
 				virtual ~FaceConverterT()
 				{
 				}
 
-				void convertIfcSurface(const std::shared_ptr<typename IfcEntityTypesT::IfcSurface>& surface,
-					const carve::math::Matrix& pos,
-					std::shared_ptr<carve::input::PolylineSetData>& polyline_data)
-				{
-					double length_factor = UnitConvert()->getLengthInMeterFactor();
+				// WHERE to put length factor?		
+				double lengthFactor = UnitConvert()->getLengthInMeterFactor();
 
-					/*	Faceconverter.h
-						For IFC4x1:
+				/*! \brief Converts \c IfcSurface to a according to subtype.
+					\param		surface	\c IfcSurface entity to be interpreted.
+					\param		pos
+					\param		polylineData
+					\return
+					\note		The \c IfcSurface subtypes are \c IfcBoundedSurface,  \c IfcElementarySurface and  \c IfcSweptSurface.
+				*/
 
-						IfcSurface
+				void convertIfcSurface(const EXPRESSReference<typename IfcEntityTypesT::IfcSurface>& surface,
+					const carve::math::Matrix& pos,
+					EXPRESSReference<carve::input::PolylineSetData>& polylineData) {

```

Comment:
```suggestion
					std::shared_ptr<carve::input::PolylineSetData>& polylineData) 
				{
```
Carve stuff cannot be EXPRESSreference, only IFC stuff.

Reply:
Also, make all functions `const throw(...)`
```suggestion
					std::shared_ptr<carve::input::PolylineSetData>& polylineData) const throw(...)
				{
```